aloe/tools.py

```diff
@@ -93,3 +95,54 @@ def guess_types(data):  # pylint:disable=too-complex
 
     #  give up
     return data
+
+
+def hook_not_reentrant(func):
+    """
+    Decorate a hook as unable to be reentered while it is already in the
+    stack.
+
+    Any further attempts to enter the hook before exiting will be replaced
+    by a no-op.
+
+    This is generally useful for step hooks where a step might call
+    :meth:`Step.behave_as` and trigger a second level of step hooks i.e.
+    when displaying information about the running test.
+    """
+
+    # pylint:disable=protected-access
+    func._entered = False
```

Comment:
Just occurred to me that this can be:

```python
entered = [False]

...

entered[0] = True
```

I.e. store this in the closure instead of the function.